// shared-backend/dist/services/ExpressAuthService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressAuthService = void 0;
class ExpressAuthService {
    constructor(expressApiUrl) {
        this.expressApiUrl = expressApiUrl;
        this.baseUrl = `${this.expressApiUrl}/auth`;
    }
    async sendMagicLink(email, redirectUrl) {
        const callbackUrl = `${this.baseUrl}/callback`;
        ;
        console.log("Sending signin with callback url:", callbackUrl);
        const res = await fetch(`${this.baseUrl}/signin`, {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({ email, callbackUrl }),
            credentials: "include",
        });
        const data = await res.json();
        if (!res.ok)
            throw new Error(data.error ?? "Failed to sign in");
    }
    async getSession() {
        const res = await fetch(`${this.baseUrl}/user`, {
            method: "GET",
            credentials: "include",
        });
        if (!res.ok)
            return null;
        const data = await res.json();
        return {
            access_token: null,
            refresh_token: null,
            user: data.user ?? null,
        };
    }
    async getUser() {
        const session = await this.getSession();
        return { user: session?.user ?? null };
    }
    async signOut() {
        const res = await fetch(`${this.baseUrl}/signout`, {
            method: "POST",
            credentials: "include",
        });
        if (!res.ok) {
            const data = await res.json();
            throw new Error(data.error ?? "Failed to sign out");
        }
    }
}
exports.ExpressAuthService = ExpressAuthService;
